weston-ivi-shell/src/ivi-controller.c

```diff
@@ -900,18 +900,21 @@ set_bkgnd_surface_prop(struct ivishell *shell)
     dest_rect.width = dest_width;
     dest_rect.height = dest_height;
 
-    calc_trans_matrix(&source_rect, &dest_rect,
-                      &shell->bkgnd_transform.matrix);
-    weston_matrix_translate(&shell->bkgnd_transform.matrix, x, y, 0.0f);
-
-    weston_log("set_bkgnd_surface_prop: x:%d y:%d s_width:%d s_height:%d d_width:%d d_height:%d\n",
-               x, y, src_width, src_height, dest_width, dest_height);
-
-    wl_list_insert(&view->geometry.transformation_list,
-                   &shell->bkgnd_transform.link);
-    view->transform.dirty = 1;
-    weston_view_update_transform(view);
-    ivi_weston_surface_schedule_repaint(w_surface);
+    /*Only repaint when size of destination bigger than 0*/
+    if (dest_rect.width != 0 && dest_rect.height != 0) {
```

Comment:
When destination of background view is 0 then it should ideally be removed from the view list OR the transform should be set in a way that end composition result produces no background view.

Reply:
Ideas is good, Harsha. I consider this, there are two views from me.
- background view only add to view list of background layer, controller by ivi-controller. I don't think we need to repeat the removing/adding the view, the background layer only one view.
-  I understand the transformation of view is calculating the position and set output to view. We can ignore the calculating, and only set no output to view. But now, there are no output, then we can't request to repaint. When new output is plugged, this function call again, then it can draw the background like our expectation.

Reply:
https://gitlab.freedesktop.org/wayland/weston/-/blob/12.0/ivi-shell/ivi-layout.c?ref_type=heads#L608